### DISCUSSION TITLE
Computed property in Livewire\Form

### DISCUSSION THREAD

**Discussion by thehands79** [livewire/livewire#7800] [1. Help] [answered]
This looks like a bug but I may have misunderstood something or made a mistake. 

If I add a Computed property to my Livewire\Form and try to call it from blade, I get an error: 'Undefined property'.

**Example**
```
use Livewire\Form;
use Livewire\Attributes\Computed;

class AdvertForm extends Form
{

    #[Computed]
    public function allTags(): Collection
    {
        return Tag::orderBy('name')->get();
    }

}

//==========

use App\Livewire\Forms\AdvertForm;

class CreateAdvert extends Component
{
    public AdvertForm $form;

    // more stuff
}

//==========

use App\Livewire\Forms\AdvertForm;

class UpdateAdvert extends Component
{
    public AdvertForm $form;

    // more stuff
}

//==========

create-or-update-advert.blade.php:

// causes 'Undefined property'
@foreach ($this->form->allTags as $tag)
    // show tag code
@endforeach

// $this->form->allTags() works although docs suggest no brackets needed
@foreach ($this->form->allTags() as $tag)
    // show tag code
@endforeach
```
![Screenshot 2024-01-25 at 07 37 54](https://github.com/livewire/livewire/assets/1532964/68ec7109-91c4-49ac-9135-90f21efe248e)
[no brackets needed in Basic Usage](https://livewire.laravel.com/docs/computed-properties#basic-usage)

**Workaround**
```
use Livewire\Form;
use Livewire\Attributes\Computed;

class CreateAdvert extends Component
{
    public AdvertForm $form;

    #[Computed]
    public function allTags(): Collection
    {
        return $this->form->allTags();
    }
}

//=========

use Livewire\Form;
use Livewire\Attributes\Computed;

class UpdateAdvert extends Component
{
    public AdvertForm $form;

    #[Computed]
    public function allTags(): Collection
    {
        return $this->form->allTags();
    }
}

//=========

class AdvertForm extends Form
{
    public function allTags(): Collection
    {
        return Tag::orderBy('name')->get();
    }
}
```

**Comment by jj15asmr** [comment, marked as answer]
I don't believe they will work in Form objects. I recently dug into the code for Computed Properties and from what I could understand, their internal implementation requires an actual component (that extends `Livewire\Component`, form objects extend `Livewire\Form`) to function properly.

**Reply by calebporzio** [reply]
This is correct, computed properties aren't supported on form objects.

**Reply by thehands79** [reply]
Thought this would be useful in the docs so created a PR for it:
https://github.com/livewire/livewire/pull/7804

**Comment by thehands79** [comment]
Follow up question: are Locked properties supported on Livewire\Form?

**Reply by jj15asmr** [reply]
I just tested and looks like they do:

![image](https://github.com/livewire/livewire/assets/48543796/b0501957-c21b-411b-8b24-cd7522f12d4f)
